include/bluetooth/scan.h

```diff
@@ -515,6 +515,65 @@ void bt_scan_filter_remove_all(void);
  */
 int bt_scan_params_set(struct bt_le_scan_param *scan_param);
 
+/**@brief Function for generic adding a new device to the connection attempts
+ *        filter.
+ *
+ * @details Use this function to add a new device to the connection attempts
+ *          filter. This filter limits a connection establish tries to
+ *          the device during one scanning session. If the device is already
+ *          added to filter array this function return with success.
+ *
+ * @param[in] addr Device address for filtering.
+ * @param[in] force Force device adding. If filtered device array is full
+ *                  overwrite the oldest device with the new one.
+ *
+ * @return 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
+int bt_scan_attempts_filter_device_generic_add(const bt_addr_le_t *addr,
```

Comment:
Maybe we could keep this function static and remove it from API?
Alternatively we could keep this function and remove remaining two. Currently we have "duplicated API".

Reply:
Done